src/intro_gui.cpp

```diff
@@ -33,13 +39,207 @@
+	AlignmentV align_v;  ///< Vertical alignment.
+
+	IntroGameViewportCommand() : command_index(0), vehicle(INVALID_VEHICLE), delay(0), zoom_adjust(0), pan_to_next(false), align_h(CENTRE), align_v(MIDDLE)
+	{
+	}
+
+	/**
+	 * Calculate effective position.
+	 * This will update the position field if a vehicle is followed.
+	 * @param vp Viewport to calculate position for.
+	 * @return Calculated position in the viewport.
+	 */
+	Point PositionForViewport(const Viewport *vp)
+	{
+		if (this->vehicle != INVALID_VEHICLE) {
+			const Vehicle *v = Vehicle::Get(this->vehicle);
+			this->position = RemapCoords(v->x_pos, v->y_pos, v->z_pos);
+		}
+
+		Point p;
+		switch (this->align_h) {
+			case LEFT: p.x = this->position.x; break;
+			case CENTRE: p.x = this->position.x - vp->virtual_width / 2; break;
+			case RIGHT: p.x = this->position.x - vp->virtual_width; break;
+		}
+		switch (this->align_v) {
+			case TOP: p.y = this->position.y; break;
+			case MIDDLE: p.y = this->position.y - vp->virtual_height / 2; break;
+			case BOTTOM: p.y = this->position.y - vp->virtual_height; break;
+		}
+		return p;
+	}
+};
+
+
 struct SelectGameWindow : public Window {
+	/** Vector of viewport commands parsed. */
+	std::vector<IntroGameViewportCommand> intro_viewport_commands;
+	/** Index of currently active viewport command. */
+	size_t cur_viewport_command_index;
+	/** Time spent (milliseconds) on current viewport command. */
+	uint cur_viewport_command_time;
+
+	/**
+	 * Find and parse all viewport command signs.
+	 * Fills the intro_viewport_commands vector and deletes parsed signs from the world.
+	 */
+	void ReadIntroGameViewportCommands()
+	{
+		intro_viewport_commands.clear();
+
+		/* Regular expression matching the commands: T, spaces, integer, spaces, flags, spaces, integer */
+		const char *sign_langauge = "^T\\s*([0-9]+)\\s*([-+A-Z0-9]+)\\s*([0-9]+)";
+		std::regex re(sign_langauge, std::regex_constants::icase);
+
+		/* List of signs successfully parsed to delete afterwards. */
+		std::vector<SignID> signs_to_delete;
+
+		for (const Sign *sign : Sign::Iterate()) {
+			std::smatch match;
+			if (std::regex_search(sign->name, match, re)) {
+				IntroGameViewportCommand vc;
+				/* Sequence index from the first matching group. */
+				vc.command_index = std::stoi(match[1].str());
+				/* Sign coordinates for positioning. */
+				vc.position = RemapCoords(sign->x, sign->y, sign->z);
+				/* Delay from the third matching group. */
+				vc.delay = std::stoi(match[3].str()) * 1000; // milliseconds
+
+				/* Parse flags from second matching group. */
+				enum IdType {
+					ID_NONE, ID_VEHICLE
+				} id_type = ID_NONE;
+				for (char c : match[2].str()) {
+					if (isdigit(c)) {
+						if (id_type == ID_VEHICLE) {
+							vc.vehicle = vc.vehicle * 10 + (c - '0');
+						}
+					} else {
+						id_type = ID_NONE;
+						switch (toupper(c)) {
+							case '-': vc.zoom_adjust = +1; break;
+							case '+': vc.zoom_adjust = -1; break;
```

Comment:
Just to confirm: it is correct that `+` maps to `-1`? And you really meant to write `+1`?

Reply:
Yes, `+` flag in the sign causes it to zoom in, and that happens to require the opposite sign for the in-game adjustment. The `+1` is just for symmetry with the `-1`. Conceptually the `vc.zoom_adjust` value should be added to the `_gui_zoom` value, which is what the code in `FixTitleGameZoom(zoom_adjust)` does.